### PR TITLE
[FIX] hw_drivers: reconnect WS on server disconnection

### PR DESCRIPTION
**Pull request by lse-odoo**
To reproduce:
 1. Sync the IoT with a 17 database
 2. Once it is set up, force shut down the server by Ctrl+C 2 times
 3. Wait a few seconds:
=> crash of `WebsocketClient` thread with the following traceback:
```
Traceback (most recent call last):
  File "/usr/lib/python3.11/threading.py", line 1038, in _bootstrap_i>
self.run()
  File "/home/pi/odoo/addons/hw_drivers/websocket_client.py", line 83>
self.ws.run_forever()
  File "/usr/local/lib/python3.11/dist-packages/websocket/_app.py", l>
raise WebSocketException("socket is already opened")
websocket._exceptions
WebSocketException
socket is already opened
```

The IoT synchronised servers can stop in 2 ways that we need to handle:
 - Gracefully (Ctrl+C):
  In this case a disconnection signal is sent to the IoT-box
  The websocket is properly closed, but it needs to be established a new connection when the server will be back.
  Solution: `while True` loop as `run_forever` will return on close. `time.sleep` for the reconnection delay

 - Forced/killed (Ctrl+C 2 times):
  In this case there is no disconnection signal received
  Solution: use `reconnect` that will automatically take care re-attempting a connection

  This will also happen with the graceful quit as `reconnect` will trigger if the server is offline while attempting the new connection

In both case, we perform a reconnection attempt with a delay of 10 seconds to avoid spamming the database

After this commit:
 The websocket will reconnect itself automatically after 10 seconds if the connection is disrupted

opw-3612528

### CODE REVIEW

**Comment by qle-odoo** [comment]
time library is no more used, so you can remove it

**Review by qle-odoo — APPROVED** [review]

**Comment by viforget** [comment]
The while 1 is necessary in the case of a proper close of the ws connection, but reconnect=10 is indeed necessary for forced shutdown

**Comment by Xavier-Do** [comment]
@robodoo override=ci/style

**Comment by lse-odoo** [comment]
Tested on windows IoT as well, same behavior than the IoT one (and this PR fix it aw well). So good to go on both boxes!

File: addons/hw_drivers/websocket_client.py [hunk 1]
```diff
@@ -75,7 +80,24 @@ def __init__(self, url):
...
+        #
+        #   This will also happen with the graceful quit as `reconnect` will trigger if the server
+        #   is offline while attempting the new connection
+        while True:
+            try:
+                run_res = self.ws.run_forever(reconnect=10)
+                _logger.debug("websocket run_forever return with %s", run_res)
+            except Exception:
+                _logger.exception("An unexpected exception happened when running the websocket")
+            _logger.debug('websocket will try to restart in 10 seconds')
```
Comment (Ysoroko) [inline addons/hw_drivers/websocket_client.py:102]: seems unnecessary as we will either get a logger.exception or another logger.debug after this, but feel free to ignore

**Review by Ysoroko — COMMENTED** [review]
Looks good to me!

**Comment by Ysoroko** [comment]
@robodoo r+

**Pull request closed without merging**